server/base/src/main/java/org/apache/accumulo/server/iterators/ServerIteratorOptions.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.accumulo.core.spi.file.rfile.compression.NoCompression;
+import org.apache.hadoop.io.compress.Compressor;
+import org.apache.hadoop.io.compress.Decompressor;
+
+import com.google.common.base.Preconditions;
+
+public class ServerIteratorOptions {
+  static final String COMPRESSION_ALGO = "__COMPRESSION_ALGO";
+
+  private static final String NONE = new NoCompression().getName();
+
+  public interface Serializer {
+    void serialize(DataOutput dataOutput) throws IOException;
+  }
+
+  public static void compressOption(final AccumuloConfiguration config,
+      IteratorSetting iteratorSetting, String option, String value) {
+    final String algo = config.get(Property.GENERAL_SERVER_ITERATOR_OPTIONS_COMPRESSION_ALGO);
+    setAlgo(iteratorSetting, algo);
+
+    if (algo.equals(NONE)) {
+      iteratorSetting.addOption(option, value);
+    } else {
+      compressOption(config, iteratorSetting, option, dataOutput -> {
+        byte[] bytes = value.getBytes(UTF_8);
+        dataOutput.writeInt(bytes.length);
+        dataOutput.write(bytes);
+      });
+    }
+  }
+
+  public static void compressOption(final AccumuloConfiguration config,
+      IteratorSetting iteratorSetting, String option, Serializer serializer) {
+    final String algo = config.get(Property.GENERAL_SERVER_ITERATOR_OPTIONS_COMPRESSION_ALGO);
+    final CompressionAlgorithm ca = Compression.getCompressionAlgorithmByName(algo);
+    final Compressor c = ca.getCompressor();
+
+    setAlgo(iteratorSetting, algo);
+
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream(); DataOutputStream dos =
+        new DataOutputStream(ca.createCompressionStream(baos, c, 32 * 1024))) {
+      serializer.serialize(dos);
+      dos.close();
+      var val = Base64.getEncoder().encodeToString(baos.toByteArray());
+      iteratorSetting.addOption(option, val);
+    } catch (IOException ioe) {
+      throw new UncheckedIOException(ioe);
+    } finally {
+      ca.returnCompressor(c);
+    }
+  }
+
+  private static void setAlgo(IteratorSetting iteratorSetting, String algo) {
+    if (iteratorSetting.getOptions().containsKey(COMPRESSION_ALGO)) {
+      Preconditions.checkArgument(iteratorSetting.getOptions().get(COMPRESSION_ALGO).equals(algo));
+    } else {
+      iteratorSetting.addOption(COMPRESSION_ALGO, algo);
+    }
+  }
+
+  public interface Deserializer<T> {
+    T deserialize(DataInputStream dataInput) throws IOException;
+  }
+
+  public static String decompressOption(Map<String,String> options, String option) {
+    var algo = options.getOrDefault(COMPRESSION_ALGO, NONE);
+    if (algo.equals(NONE)) {
+      return options.get(option);
+    }
+
+    return decompressOption(options, option, dataInput -> {
+      int len = dataInput.readInt();
+      byte[] data = new byte[len];
+      dataInput.readFully(data);
+      return new String(data, UTF_8);
+    });
+  }
+
+  public static <T> T decompressOption(Map<String,String> options, String option,
```

Comment:
Seems like this is not used, could remove it and the related code.  Can always add it back later if there is a need.

Reply:
I was only looking at what methods TabletManagementIterator used from this class.  If its used internally, then leaving it is fine.

Reply:
Fixed in bf81226